src/middleware/etag/index.ts

```diff
@@ -57,15 +57,19 @@ function etagMatches(etag: string, ifNoneMatch: string | null) {
 export const etag = (options?: ETagOptions): MiddlewareHandler => {
   const retainedHeaders = options?.retainedHeaders ?? RETAINED_304_HEADERS
   const weak = options?.weak ?? false
-  const generator =
-    options?.generateDigest ??
-    ((body: Uint8Array) =>
-      crypto.subtle.digest(
-        {
-          name: 'SHA-1',
-        },
-        body
-      ))
+  let generator = options?.generateDigest
+
+  if (!generator) {
+    if (crypto && crypto.subtle) {
+      generator = (body: Uint8Array) =>
+        crypto.subtle.digest(
+          {
+            name: 'SHA-1',
+          },
+          body
+        )
+    }
+  }
```

Comment:
The initialisation conditions are complicated, so I think it would be better to make it a separate function.